llama_stack/providers/impls/meta_reference/agents/agents.py

```diff
@@ -138,24 +138,89 @@ async def _create_agent_turn_streaming(
+                f"Could not JSON decode turn for {turn_id}"
+            ) from e
+        try:
+            turn = Turn(**turn)
+        except Exception as e:
+            raise ValueError(
+                f"Could not validate(?) Turns for {turn_id}"
+            ) from e
+        return turn
 
     async def get_agents_step(
-        self, agent_id: str, turn_id: str, step_id: str
+        self, agent_id: str, session_id: str, turn_id: str, step_id: str
     ) -> AgentStepResponse:
-        raise NotImplementedError()
+        turn = await self.persistence_store.get(f"session:{agent_id}:{session_id}:{turn_id}")
+        try:
+            turn = json.loads(turn)
+        except json.JSONDecodeError as e:
+            raise ValueError(
+                f"Could not JSON decode turn for {turn_id}"
+            ) from e
+        try:
+            turn = Turn(**turn)
+        except Exception as e:
+            raise ValueError(
+                f"Could not validate(?) Turns for {turn_id}"
+            ) from e
+        steps = turn.steps
+        for step in steps:
+            if step.step_id == step_id:
+                return AgentStepResponse(step=step)
+        raise ValueError(f"Provided step_id {step_id} could not be found")
 
     async def get_agents_session(
         self,
         agent_id: str,
         session_id: str,
         turn_ids: Optional[List[str]] = None,
     ) -> Session:
-        raise NotImplementedError()
+        session = await self.persistence_store.get(f"session:{agent_id}:{session_id}")
+        try:
+            session = Session(**json.loads(session))
+        except json.JSONDecodeError as e:
+            raise ValueError(
+                f"Could not JSON decode session for {agent_id} and {session_id}"
+            ) from e
+        turns = []
+        if turn_ids:
+            for turn_id in turn_ids:
+                turn = await self.persistence_store.get(f"session:{agent_id}:{session_id}:{turn_id}")
+                try:
+                    turn = json.loads(turn)
+                except json.JSONDecodeError as e:
+                    raise ValueError(
+                        f"Could not JSON decode session for {agent_id} and {session_id}"
+                    ) from e
+                try:
+                    turn = Turn(**turn)
+                except Exception as e:
+                    raise ValueError(
+                        f"Could not validate(?) Turns for {turn_id}"
+                    ) from e
+                turns.append(turn)
+        return Session(
+            session_name=session.session_name,
+            session_id=session_id,
+            turns=turns if turns else [],
+            started_at=session.started_at
+        )
 
     async def delete_agents_session(self, agent_id: str, session_id: str) -> None:
-        raise NotImplementedError()
+        try:
+            await self.persistence_store.delete(f"session:{agent_id}:{session_id}")
+        except:
```

Comment:
you are catching all Exceptions here, that's not good. Can you just not catch and let the exception propagate upwards?